staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -505,6 +523,7 @@ func (c *AvailableConditionController) deleteService(obj interface{}) {
 	for _, apiService := range c.getAPIServicesFor(castObj) {
 		c.enqueue(apiService)
 	}
+	delete(c.cache[castObj.Namespace], castObj.Name)
```

Comment:
this isn't correct. Add/Update/Delete events of **APIService** objects are what should modify the cache map

Reply:
Considering the getAPIServicesFor outline below, should the delete service event remove the cache entry ?
If not, where is the proper place to remove cache entry ?

Reply:
> should the delete service event remove the cache entry ?

no

> If not, where is the proper place to remove cache entry ?

when an APIService object is updated or deleted

Reply:
APIService objects change very rarely. Whenever one changes (gets added/updated/removed), we can rebuild the cache.